tests/Bridge/Symfony/Routing/ApiLoaderTest.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ApiPlatform\Core\Tests\Symfony\Bridge\Bundle\DependencyInjection;
+
+use ApiPlatform\Core\Bridge\Symfony\Routing\ApiLoader;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceNameCollectionFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\ResourceMetadata;
+use ApiPlatform\Core\Metadata\Resource\ResourceNameCollection;
+use ApiPlatform\Core\Tests\Fixtures\DummyEntity;
+use Prophecy\Argument;
+use Symfony\Component\HttpKernel\KernelInterface;
+use Symfony\Component\Routing\Route;
+
+/**
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+class ApiLoaderTest extends \PHPUnit_Framework_TestCase
+{
+    private function getApiLoaderWithResourceMetadata(ResourceMetadata $resourceMetadata)
+    {
+        $routingConfig = __DIR__.'/../../../../src/Bridge/Symfony/Bundle/Resources/config/routing';
+
+        $kernelProphecy = $this->prophesize(KernelInterface::class);
+        $kernelProphecy->locateResource(Argument::any())->willReturn($routingConfig);
+
+        $resourceMetadataFactoryProphecy = $this->prophesize(ResourceMetadataFactoryInterface::class);
+        $resourceMetadataFactoryProphecy->create(DummyEntity::class)->willReturn($resourceMetadata);
+
+        $resourceNameCollectionFactoryProphecy = $this->prophesize(ResourceNameCollectionFactoryInterface::class);
+        $resourceNameCollectionFactoryProphecy->create()->willReturn(new ResourceNameCollection([DummyEntity::class]));
+
+        $apiLoader = new ApiLoader($kernelProphecy->reveal(), $resourceNameCollectionFactoryProphecy->reveal(), $resourceMetadataFactoryProphecy->reveal());
+
+        return $apiLoader;
+    }
+
+    public function testApiLoader()
+    {
+        $resourceMetadata = new ResourceMetadata();
+
+        $resourceMetadata = $resourceMetadata->withShortName('dummy');
+
+        //default operation based on OperationResourceMetadataFactory
+        $resourceMetadata = $resourceMetadata->withItemOperations([
+            'get' => ['method' => 'GET'],
+            'put' => ['method' => 'PUT'],
+            'delete' => ['method' => 'DELETE'],
+        ]);
+
+        //custom operations
+        $resourceMetadata = $resourceMetadata->withCollectionOperations([
+            'my_op' => ['method' => 'GET', 'controller' => 'some.service.name'], //with controller
+            'my_second_op' => ['method' => 'POST'], //without controller, takes the default one
```

Comment:
@teohhanhui this test for example. Here the controller will default to `api_platform.action.post_item`. IMO it's an expected behavior and this is the same behavior as in 1.x.